languages/rust/generic/Parse_rust_tree_sitter.ml

```diff
@@ -1194,11 +1194,14 @@ and map_expression (env : env) (x : CST.expression) =
       in
       let rhs = map_expression env v3 in
       G.AssignOp (lhs, (op, tok), rhs)
-  | `Type_cast_exp (v1, v2, v3) ->
+  | `Type_cast_exp (v1, v2, v3) -> (
       let expr = map_expression env v1 in
       let as_ = token env v2 (* "as" *) in
       let type_ = map_type_ env v3 in
-      G.Cast (type_, as_, expr)
+      match expr with
+      | { e = N (Id ((n, t), _)); _ } when AST_generic.is_metavar_name n ->
```

Comment:
Could we case on the `env` (which may be `Pattern` or `Target`, defined above) to ensure that this only happens during pattern parsing? You can consult `Parse_julia_tree_sitter` for an example of where we do this.